### PR TITLE
MAUT-11617 : Campaign -> Condition -> Custom Object Field values : Date picker unavailable for Date type fields

### PR DESCRIPTION
**Pull request by rahuld-dev**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | No
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | N/A 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | https://acquia.atlassian.net/browse/MAUT-11617

#### Description:

Adding date-picker for date and datetime fields in Campaign -> Condition -> Custom Object Field values.

#### Steps to test this PR:

1. Open this PR for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create Custom Object with date and datetime type field
3. Navigate to Campaign →  Condition →  Contact Field values
4. Select date/datetime type field
5. Verify that date-picker should be available for date and datetime field value.
image

### CODE REVIEW

**Review by rohitp19 — DISMISSED** [review]

File: Assets/js/custom-objects.js [hunk 1]
```diff
@@ -46,6 +46,7 @@ CustomObjects = {
             'autocomplete': valueField.attr('autocomplete'),
             'value': valueField.attr('value')
         };
+        let fieldType = selectedField.attr('data-field-type');
```
Comment (fedys) [inline Assets/js/custom-objects.js]:
```suggestion
        const fieldType = selectedField.attr('data-field-type');
```
Let's use `const` for variables that are never written to.

**Review by fedys — CHANGES_REQUESTED** [review]
Thanks @rahuld-dev! Please see my code suggestion.

**Review by fedys — DISMISSED** [review]
Thanks @rahuld-dev!

**Comment by fedys** [comment]
@rohitp19 you can start your code review now.

**Comment by rohitp19** [comment]
@rahuld-dev It looks like it is not deployed to beta correctly. Can you please check?

**Review by rohitp19 — DISMISSED** [review]
Thanks @rahuld-dev Tested on https://rohitaws1.qa.acs.acquia.io/.

**Review by fedys — APPROVED** [review]
Thanks @rahuld-dev!

**Comment by fedys** [comment]
@rohitp19 you can start your code review.

**Review by avikarshasaha — APPROVED** [review]
Tested in beta! Thanks @rahuld-dev

**Pull request merged**